doc/changes.md

```diff
@@ -1,5 +1,29 @@
 # Changelog
 
+## Release v0.14.1 (2025/01/30)
+
+[![Digital Object Identifier for PyGMT v0.14.1](https://zenodo.org/badge/DOI/10.5281/zenodo.14742338.svg)](https://doi.org/10.5281/zenodo.14742338)
+
+### Highlights
```

Comment:
I am wondering if we need the highlights sections for this release. Or should we add the datetime-bug-fix as highlight?